Mathlib/Combinatorics/Additive/NoDoubling.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Patrick Luo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Patrick Luo
+-/
+import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+# Sets with no doubling
+
+This file characterises sets with no doubling (finsets `A` such that `#(A ^ 2) = #A`) as the sets
+which are either empty or translates of a subgroup.
+-/
+
+open MulOpposite MulAction
+open scoped Pointwise RightActions
+
+namespace Finset
+variable {G : Type*} [Group G] [DecidableEq G] {A : Finset G}
+
+/-- A set with no doubling is either empty or the translate of a subgroup. -/
+@[to_additive "A set with no doubling is either empty or the translate of a subgroup."]
+lemma exists_subgroup_of_no_doubling (hA : #(A * A) ≤ #A) :
```

Comment:
Can we relate this hypothesis to one involving the doubling constant? I don't mind if it's in this file or in the doubling constant one (or even in a different PR), but we should have the statement d(A) = 1 iff #(A * A) <= #A somewhere

Reply:
I have this lemma proved locally in LeanCamCombi, but I was wondering whether it should live here or in the doubling constant file. We could just leave it to another PR for now, maybe

Reply:
Yeah that's fine, let's just add a TODO for it. I don't have a strong preference where it should live, although this file should reference it if it doesn't contain it.